controllers/aggregated_clusteroperator_controller.go

```diff
@@ -62,40 +64,51 @@ func (r *AggregatedClusterOperatorReconciler) Reconcile(ctx context.Context, req
-	coBuilder.WithAvailable(configv1.ConditionFalse, "", "")
+	// TODO: always set a reason (message is optional, but desirable)
+	coBuilder.WithProgressing(metav1.ConditionTrue, "", "")
+	coBuilder.WithDegraded(metav1.ConditionFalse, "", "")
+	coBuilder.WithAvailable(metav1.ConditionFalse, "", "")
 	coBuilder.WithVersion("operator", r.ReleaseVersion)
-	coBuilder.WithRelatedObject("", "namespaces", "", r.SystemNamespace)
-	coBuilder.WithRelatedObject("platform.openshift.io", "platformoperators", "", "")
-	coBuilder.WithRelatedObject("core.rukpak.io", "bundles", "", "")
-	coBuilder.WithRelatedObject("core.rukpak.io", "bundledeployments", "", "")
+	coBuilder.WithRelatedObject(configv1.ObjectReference{Group: "", Resource: "namespaces", Name: r.SystemNamespace})
+
+	// NOTE: Group and resource can be referenced without name/namespace set, which is a signal
+	// that _ALL_ objects of that group/resource are related objects. This is useful for
+	// must-gather automation.
+	coBuilder.WithRelatedObject(configv1.ObjectReference{Group: platformv1alpha1.GroupName, Resource: "platformoperators"})
+
+	// TODO: move platform operator ownership of rukpak objects out prior to rukpak or PO GA.
+	coBuilder.WithRelatedObject(configv1.ObjectReference{Group: rukpakv1alpha1.GroupVersion.Group, Resource: "bundles"})
+	coBuilder.WithRelatedObject(configv1.ObjectReference{Group: rukpakv1alpha1.GroupVersion.Group, Resource: "bundledeployments"})
 
 	poList := &platformv1alpha1.PlatformOperatorList{}
 	if err := r.List(ctx, poList); err != nil {
 		return ctrl.Result{}, err
 	}
 	if len(poList.Items) == 0 {
 		// No POs on cluster, everything is fine
-		coBuilder.WithAvailable(configv1.ConditionTrue, "No POs are present in the cluster", "NoPOsFound")
-		coBuilder.WithProgressing(configv1.ConditionFalse, "No POs are present in the cluster")
+		// TODO: cleanup condition reasons.
+		//   1. use constants for condition reasons.
+		//   2. discuss/agree on reason values.
+		//   3. don't use abbreviations.
+		coBuilder.WithAvailable(metav1.ConditionTrue, "NoPOsFound", "No POs are present in the cluster")
+		coBuilder.WithProgressing(metav1.ConditionFalse, "", "No POs are present in the cluster")
 		return ctrl.Result{}, nil
 	}
 
 	// check whether any of the underlying PO resources are reporting
 	// any failing status states, and update the aggregate CO resource
 	// to reflect those failing PO resources.
 	if statusErrorCheck := util.InspectPlatformOperators(poList); statusErrorCheck != nil {
-		coBuilder.WithAvailable(configv1.ConditionFalse, statusErrorCheck.Error(), "POError")
+		coBuilder.WithAvailable(metav1.ConditionFalse, "POError", statusErrorCheck.Error())
 		return ctrl.Result{}, nil
 	}
-	coBuilder.WithAvailable(configv1.ConditionTrue, "All POs in a successful state", "POsHealthy")
-	coBuilder.WithProgressing(configv1.ConditionFalse, "All POs in a successful state")
+	coBuilder.WithAvailable(metav1.ConditionTrue, "POsHealthy", "All POs in a successful state")
```

Comment:
I've seen a bunch of other core ClusterOperator resources set the "AsExpected" reason during success states.